### ISSUE TITLE
Package builds on main have version 0.0.0

### ISSUE THREAD

**Issue by samtygier-stfc** [mantidproject/mantidimaging#1388]
### Summary

https://anaconda.org/mantid/mantidimaging/files is showing a linux-64/mantidimaging-0.0.0-py38_1.tar.bz2 package built on 2022-03-29 and no other recent unstable builds. So probably the version number is not being set correctly in the builds from main.

After this mornings merge, the previous 0.0.0 package is gone and there is a new one with the date today.

### Failure Logs

From the Actions build log
```
Extracting conda package attributes for upload
Creating package "mantidimaging"
Creating release "0.0.0"
Uploading file "mantid/mantidimaging/0.0.0/linux-64/mantidimaging-0.0.0-py38_1.tar.bz2"
Warning:  Distribution "linux-64/mantidimaging-0.0.0-py38_1.tar.bz2" already exists. Removing.
```

**Comment by samtygier-stfc** [comment]
In the older logs the version gets picked up from `git describe` here (for some reason the output shows before the commands)

```
2.4.0a1-58-g67d20792

On branch main
Your branch is up to date with 'origin/main'.

nothing to commit, working tree clean

checkout: 'HEAD'
==> /usr/bin/git log -n1 <==

==> /usr/bin/git describe --tags --dirty <==

==> /usr/bin/git status <==
```

first bad build is:

```
 commit b4e79be6bee1f98eb699a41de2ba1af8c258ab7b
Author: rbauststfc <38210467+rbauststfc@users.noreply.github.com>
Date:   Fri Mar 11 12:54:04 2022 +0000

    Merge pull request #1368 from mantidproject/1338-actions-docs
    
    Move doc building into its own workflow

On branch main
Your branch is up to date with 'origin/main'.

nothing to commit, working tree clean

checkout: 'HEAD'
==> /usr/bin/git log -n1 <==

==> /usr/bin/git describe --tags --dirty <==

==> /usr/bin/git status <==
```

I suspect its removing `fetch-depth: 0` in #1368 that has caused the issue. Before we checked out the entire git history, not just the HEAD.

**Comment by samtygier-stfc** [comment]
https://github.com/actions/checkout/issues/204

`fetch-refs` might help

**Comment by samtygier-stfc** [comment]
That built and published mantidimaging-2.4.0a1_108-py38_1.tar.bz2, so it looks like it worked. 
https://anaconda.org/mantid/mantidimaging/files
